### PR TITLE
Modifying the toolset compiler to support Dev15.

### LINKED ISSUE #11994: Toolset Compilers do not work properly in Dev15
`build/NuGetAdditionalFiles/Microsoft.Net.Compilers.props` is currently hard-coded to only be used if `MSBuildToolsVersion` is `14.0`. However, in Dev15 the `MSBuildToolsVersion` is `15.0`.

The file should, at the very least, be updated to support both `14.0` and `15.0`.

### PR DESCRIPTION
**Pull request by tannergooding**
This resolves #11994. FYI. @dotnet/roslyn-compiler

### CODE REVIEW

**Comment by tannergooding** [comment]
test windows_eta_open_prtest please
test vsi please

**Comment by tannergooding** [comment]
Also FYI. @dotnet/roslyn-infrastructure

**Comment by jaredpar** [comment]
:+1: 

Wonder if there is a better way to future proof this code .

**Comment by TyOverby** [comment]
LGTM

**Comment by tannergooding** [comment]
@jaredpar, Do you want me to log a work item (to future proof this if possible or to document that this needs upgrading during major revision changes otherwise)?

**Comment by agocke** [comment]
I was trying to be conservative and not affect Dev12+earlier, but I'm not sure it's necessary.

These redirect points weren't there before Dev14, which probably just means that the toolset compiler won't work. But that's true with the conditional gate anyway. We can probably just remove that condition.

File: build/NuGetAdditionalFiles/Microsoft.Net.Compilers.props [hunk 1]
```diff
@@ -11,15 +11,15 @@
        new properties in the local build task. -->
   <UsingTask TaskName="Microsoft.CodeAnalysis.BuildTasks.Csc"
              AssemblyFile="$(MSBuildThisFileDirectory)..\tools\Microsoft.Build.Tasks.CodeAnalysis.dll"
-             Condition="'$(MSBuildToolsVersion)' == '14.0'" />
+             Condition="'$(MSBuildToolsVersion)' == '14.0' OR '$(MSBuildToolsVersion)' == '15.0'" />
```
Comment (AlekseyTs) [inline build/NuGetAdditionalFiles/Microsoft.Net.Compilers.props]: Could we use condition that accepts all versions greater than 14? It feels like the current condition creates a maintenance burden.

**Comment by tannergooding** [comment]
I've updated this to explicitly check for the 4 `MSBuildToolsVersions` that are pre-14.0 and error out (with a helpful message) if such a build is attempted. It should handle all the scenarios we care about and shouldn't need any update in the future.

**Comment by agocke** [comment]
@tannergooding LGTM

**Comment by tannergooding** [comment]
test windows_eta_open_prtest please

`error MSB4018: The "CreateZipPackage" task failed unexpectedly.`

**Comment by tannergooding** [comment]
test windows_eta_open_prtest please

error MSB4018: The "CreateZipPackage" task failed unexpectedly.

**Comment by tannergooding** [comment]
test windows_eta_open_prtest please

**Comment by tannergooding** [comment]
@jasonmalinowski, @dotnet/roslyn-infrastructure, ETA is still failing due to CreateZipPackage failure. I've validated that this does not fail (when manually run on the jenkins machine) and when run locally.

**Pull request merged**